### PR TITLE
Remove job_queue

### LINKED ISSUE #7885: Remove JobQueue code
Blockers:

 - [x] #7187 
 - [x] #6676 
 - [ ] Remove JobQueue SLURM driver
 - [ ] Remove JobQueue LSF driver
 - [x] #7912
 - [ ] #7985 
 - [x] #8170 
 - [x] https://github.com/equinor/ert/issues/8246
 - [x] #8200 

NB! Be aware that `--disable-scheduler` will stop working once we remove the old code, which implies that even-though users have specified LSF as queue, when running it as `test_run` with `--disable-scheduler` will want to use JobQueue LOCAL driver.

**Blocked until the feature freeze in komodo August release!**

### LINKED ISSUE #7985: Remove LOCAL JobQueue driver code
Since we are not using LOCAL JobQueue driver anymore, we should remove it.
Explanation in #7885

### PR DESCRIPTION
**Pull request by eivindjahren**
**Issue**
Resolves #7885 
Resolves #7985 

This also exposes JobStatus through the top level so that we can change the import used by everest to `from ert import JobStatus`.

Unblocks https://github.com/equinor/ert/issues/7275

It will be necessary to remove the feature toggling in everest after merging this.

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Comment by berland** [comment]
Should we add a compatibility layer for Everest in an interim period?

```
/../everest] main$ rg -i job_queue
tests/test_detached.py
8:from ert.job_queue import JobStatus

everest/suite.py
17:from ert.job_queue import JobStatus

everest/simulator/simulator.py
9:from ert.job_queue import WorkflowRunner

everest/detached/__init__.py
14:from ert.job_queue import JobStatus

everest/optimizer/ropt_plugins.py
7:from ert.job_queue import WorkflowRunner
```

**Comment by eivindjahren** [comment]
> Should we add a compatibility layer for Everest in an interim period?

I think we can perform the moves we want to do and expose these through top level. Then we can just move everest over to those.

**Comment by berland** [comment]
The readme states that "Most new code is in Python" - this can be changed to something along "_all code is in Python_".

**Comment by berland** [comment]
Hold merge until feature freeze for komodo 2024.08 has passed.

File: codecov.yml [hunk 1]
```diff
@@ -4,4 +4,4 @@ fixes:
 comment:
     # The code coverage is made up of 5 test runs so only after all coverage
```
Comment (yngve-sk) [inline codecov.yml:5]: does this relate to removing job queue? also the comment above says 5 but should be 4?

File: README.md [hunk 2]
```diff
@@ -34,9 +34,7 @@ For examples and help with configuration, see the [ert Documentation](https://er
 
 ## Developing
 
-ert was originally written in C/C++ but most new code is Python.
-
-### Developing Python
+ert was originally written in C/C++ but is now only Python.
```
Comment (xjules) [inline README.md:37]: Nice readme update!

File: codecov.yml (hunk 1 above)
Comment (eivindjahren) [reply codecov.yml:5]: Fixed

File: src/ert/__init__.py [hunk 3]
```diff
@@ -13,11 +13,11 @@
     ForwardModelStepDocumentation,
 )
 from .data import MeasuredData
-from .job_queue import JobStatus
 from .libres_facade import LibresFacade
 from .simulator import BatchSimulator, BatchContext
 from .workflow_runner import WorkflowRunner
 from .plugins import plugin
+from .scheduler import JobStatus
```
[the remark this answers is not in the document]
Comment (eivindjahren) [reply src/ert/__init__.py:20]: Created an issue for it https://github.com/equinor/ert/issues/8342

**Review by yngve-sk — APPROVED** [review]
Overall LGTM, fixup some smaller things. Make issues/smaller fixes pertaining to comments

**Pull request merged**